File: Iniciante/1021.php
<?php

$n = readline();

echo "NOTAS:\n";
echo intdiv($n,100) . " nota(s) de R$ 100.00\n";
$x = $n%100;
echo intdiv($x,50) . " nota(s) de R$ 50.00\n";
$x = $x%50;
echo intdiv($x,20) . " nota(s) de R$ 20.00\n";
$x = $x%20;
echo intdiv($x,10) . " nota(s) de R$ 10.00\n";
$x = $x%10;
echo intdiv($x,5) . " nota(s) de R$ 5.00\n";
$x = $x%5;
echo intdiv($x,2) . " nota(s) de R$ 2.00\n";
$x = $x%2;

$m = $n*100;

echo "MOEDAS:\n";
echo intdiv($x,1) . " moeda(s) de R$ 1.00\n";
$y = $m%100;
echo intdiv($y,50) . " moeda(s) de R$ 0.50\n";
$y = $y%50;
echo intdiv($y,25) . " moeda(s) de R$ 0.25\n";
$y = $y%25;
echo intdiv($y,10) . " moeda(s) de R$ 0.10\n";
$y = $y%10;
echo intdiv($y,5) . " moeda(s) de R$ 0.05\n";
$y = $y%5;
echo "$y moeda(s) de R$ 0.01\n";

?>

File: Iniciante/1021.c
#include <stdio.h>

int main()
{

    double valor;
    int n, m, x, y;

    scanf("%lf", &valor);

    m = valor * 100;
    n = (int)valor;

    printf("NOTAS:\n");
    printf("%d nota(s) de R$ 100.00\n", n / 100);
    x = n % 100;
    printf("%d nota(s) de R$ 50.00\n", x / 50);
    x = x % 50;
    printf("%d nota(s) de R$ 20.00\n", x / 20);
    x = x % 20;
    printf("%d nota(s) de R$ 10.00\n", x / 10);
    x = x % 10;
    printf("%d nota(s) de R$ 5.00\n", x / 5);
    x = x % 5;
    printf("%d nota(s) de R$ 2.00\n", x / 2);
    x = x % 2;

    printf("MOEDAS:\n");
    printf("%d moeda(s) de R$ 1.00\n", x);
    y = m % 100;
    printf("%d moeda(s) de R$ 0.50\n", y / 50);
    y = y % 50;
    printf("%d moeda(s) de R$ 0.25\n", y / 25);
    y = y % 25;
    printf("%d moeda(s) de R$ 0.10\n", y / 10);
    y = y % 10;
    printf("%d moeda(s) de R$ 0.05\n", y / 5);
    y = y % 5;
    printf("%d moeda(s) de R$ 0.01\n", y);

    return 0;
}